app/assets/javascripts/discourse/app/lib/user-menu/tab.js

```diff
@@ -45,7 +45,7 @@ export default class UserMenuTab {
   }
 
   /**
-   * @returns {string} Dasherized version of the component name that should be rendered in the panel area when the tab is active.
+   * @returns {Component} Component class that should be rendered in the panel area when the tab is active.
```

Comment:
Oh this is super nice! I assume it's now no longer possible for plugins/themes to introduce components with conflicting names?